Source/Redemption/Private/Bone.cpp

```diff
@@ -32,7 +36,20 @@ void ABone::Tick(float DeltaTime)
 
 }
 
+//this function is triggered when the bone overlaps with something
 void ABone::OnBoneBeginOverlap(UPrimitiveComponent* OverlappedComp, AActor* OtherActor, UPrimitiveComponent* OtherComp, int32 OtherBodyIndex, bool bFromSweep, const FHitResult& SweepResult)
 {
-	
+	if (OtherActor == this)
+	{
+		return;
+	}
+
+	if (OtherActor != UGameplayStatics::GetPlayerPawn(GetWorld(), 0))
```

Comment:
Is there a chance that OtherActor == nullptr?

Reply:
it has to hit something, so i don't think there is. I never saw it being nullptr